universal-application-tool-0.0.1/app/repository/ProgramRepository.java

```diff
@@ -159,40 +164,93 @@ public Program createOrUpdateDraft(Program existingProgram) {
     return database
         .find(Program.class)
         .where()
-        .eq(
-            "name",
-            database.find(Program.class).setId(programId).select("name").findSingleAttribute())
+        .in("name", database.find(Program.class).select("name").where().eq("id", programId).query())
+        .query()
         .findList()
         .stream()
         .collect(ImmutableList.toImmutableList());
   }
 
-  public ImmutableList<Long> getAllProgramVersionIds(long programId) {
+  public PaginationResult<Application> getApplicationsForAllProgramVersions(
+      long programId, PaginationSpec paginationSpec, Optional<String> searchNameFragment) {
+    ExpressionList<Application> query =
+        database
+            .find(Application.class)
+            .fetch("program")
+            .orderBy("id desc")
+            .where()
+            .in("program_id", allProgramVersionsQuery(programId))
+            .in(
+                "lifecycle_stage",
+                ImmutableList.of(LifecycleStage.ACTIVE, LifecycleStage.OBSOLETE));
+
+    if (searchNameFragment.isPresent() && !searchNameFragment.get().isBlank()) {
+      String search = searchNameFragment.get();
+
+      if (search.matches("^\\d+$")) {
```

Comment:
This variation in behavior (id vs name) is undocumented and kinda brittle.  I could in theory name my program with a number and this would never be able to find it.  As well for EG ID "14" this will return the application with id 14 and applications for applicant 14 which doesn't seem correct.

It's not clear to me why we're supporting searching on DB IDs which is an implementation detail and not part of a User Journey that I know of.

Are Admins expected to be able to work with Applicants based on their DB ID'd?

Are Admins expected to lookup programs by their IDs?

Reply:
Yeah it's requested behavior in the linked issue: "They need to be able to search for applications using the Application ID and/or applicant's name."

Reply:
The code also include applicant_id which is not in that statement, is that a requirement?

This level of search is often going to return noise as application and applicant IDs will overlap a lot.  So seemingly one is going to be a noisy result for the other in most situations.

It's likely fine for now... but it's debt I think as it's unexpected and confusing for non SEA users.

Reply:
I included applicant_id because I thought I remembered Elise mentioning it but I'm not finding that anywhere now. Let's just do application ID and we can easily add applicant ID later if desired.